### PR TITLE
docs: massive refactor to improve readibility and description

### PR DESCRIPTION
**Pull request by giuliowaitforitdavide**

### CODE REVIEW

File: recsyslearn/dataset/segmentations.py
Comment (mmosc) [inline recsyslearn/dataset/segmentations.py:1]: I think it is very important to mention that the interaction segmentation is cumulative / in terms of quantiles, while the activity segmentation is not. And also in the documentation.

File: CONTRIBUTING.md
[the remark this answers is not in the document]
Comment (giuliowaitforitdavide) [reply CONTRIBUTING.md:1]: Here I have to ask where do you want me to fix this

File: README.rst
[the remark this answers is not in the document]
Comment (giuliowaitforitdavide) [reply README.rst:1]: I'm pushing the fixes, I solved other conversation where you told me the resolution already

File: recsyslearn/dataset/segmentations.py
Comment (giuliowaitforitdavide) [reply recsyslearn/dataset/segmentations.py:1]: Let me know if it's godd

File: CONTRIBUTING.md
[the remark this answers is not in the document]
Comment (giuliowaitforitdavide) [reply CONTRIBUTING.md:1]: Ok here I have some doubts:
1. Where can I find the citation?
2. I have to admit it's a little bit difficult. Also, when choosing [bug report](https://github.com/giuliowaitforitdavide/recsyslearn/issues/new/choose) (yes, I also fixed the link in the file so one can easily go to this page lol) there's already a template for this.
3. Fixing with next push
4. How would you fix this? In this push I'm also fixing the template issue link to suggest enhancements
5. I removed this tool, I used a CONTRIBUTING.md generator and I forgot to remove broken links and suggestions which I honestly don't like
6. Fixing with next push
7. Same as 5.

File: recsyslearn/dataset/segmentations.py
Comment (giuliowaitforitdavide) [reply recsyslearn/dataset/segmentations.py:1]: I specified cumulative just for interactionSegmentation

Comment (mmosc) [reply recsyslearn/dataset/segmentations.py:1]: I am a bit confused...the two docstrings say:
```
class ActivitySegmentation(Segmentation):

    """
    Segmentation of users based on their number of interaction.
    """
```
and 
```
class InteractionSegmentation(Segmentation):

    """
    Segmentation of items based on the number of interaction they have.
    """
```

So the two docstrings are the same (and therefore the documentation, also?).

**Review by mmosc — APPROVED** [review]
Thanks for the massive update :) Looks good to me!

**Pull request merged**